### PR TITLE
Brings over history on DI.Specification.Tests from Extensions

### PR DESCRIPTION
**Pull request by maryamariyan**
History on:
https://github.com/dotnet/extensions/commits/master/src/DependencyInjection/DI.Specification.Tests

DependencyInjections tests need this:
https://github.com/dotnet/runtime/pull/33563#issuecomment-598981289

### CODE REVIEW

**Comment by ericstj** [comment]
Why are these under libraries/Common/tests rather than next to the DI library?  Are they shared between a number of test projects?

**Comment by ericstj** [comment]
Side note: we should probably rename the area label "area-AspNet-Extensions" to "area-Microsoft.Extensions"

**Review by ericstj — APPROVED** [review]
Commits look fine, please double check the path is what you intended.

**Comment by maryamariyan** [comment]
changed path to:
src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Specification.Tests/

File: src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Specification.Tests/Microsoft.Extensions.DependencyInjection.Specification.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <Description>Suite of xUnit.net tests to check for container compatibility with Microsoft.Extensions.DependencyInjection.</Description>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (ericstj) [inline src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Specification.Tests/Microsoft.Extensions.DependencyInjection.Specification.Tests.csproj]:
```suggestion
    <TargetFrameworks>$(NetCoreAppCurrent);$(NetFrameworkCurrent)</TargetFrameworks>
```

File: src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Specification.Tests/Microsoft.Extensions.DependencyInjection.Specification.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      This intentionally does not reference 'xunit', 'xunit.core', or any runner packages.
+      XUnit recommends only using xunit.extensibility.*, xunit.assert, and xunit.abstractions for packages which "extend" xunit.
+      This allows consumers to decide which type of xunit runner they want to use to run these tests,
+      and avoids problems with `dotnet pack`.
+
+      See https://xunit.github.io/docs/nuget-packages and the special note in https://xunit.github.io/releases/2.3.
+    -->
+    <Reference Include="xunit.assert" />
+    <Reference Include="xunit.extensibility.core" />
+  </ItemGroup>
```
Comment (ericstj) [inline src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Specification.Tests/Microsoft.Extensions.DependencyInjection.Specification.Tests.csproj]:
```suggestion

```

**Comment by ericstj** [comment]
@maryamariyan for some reason this is trying to build DI.Specification.Tests.  Do you know why that would be?  Maybe if we don't need this CSProj we just delete it.

**Review by ericstj — APPROVED** [review]

**Pull request merged**